platform/ios/src/MGLMapboxEvents.m

```diff
@@ -433,51 +434,63 @@ - (MGLMapboxEventAttributes *)fullyFormedEventForEvent:(NSString *)event withAtt
-                                                     MGLEventKeyAccessibilityFontScale: @([self contentSizeScale]),
-                                                     MGLEventKeyOrientation: [self deviceOrientation],
-                                                     MGLEventKeyWifi: @([[MGLReachability reachabilityForLocalWiFi] isReachableViaWiFi])} mutableCopy];
-    [self addBatteryStateToAttributes:attributes];
+    MGLMutableMapboxEventAttributes *attributes = [NSMutableDictionary dictionary];
+    [self updateDictionary:attributes withKey:MGLEventKeyEvent value:MGLEventTypeMapLoad];
+    [self updateDictionary:attributes withKey:MGLEventKeyCreated value:[self.rfc3339DateFormatter stringFromDate:[NSDate date]]];
+    [self updateDictionary:attributes withKey:MGLEventKeyVendorID value:self.data.vendorId];
+    [self updateDictionary:attributes withKey:MGLEventKeyModel value:self.data.model];
+    [self updateDictionary:attributes withKey:MGLEventKeyOperatingSystem value:self.data.iOSVersion];
+    [self updateDictionary:attributes withKey:MGLEventKeyResolution value:@(self.data.scale)];
+    [self updateDictionary:attributes withKey:MGLEventKeyAccessibilityFontScale value:@([self contentSizeScale])];
+    [self updateDictionary:attributes withKey:MGLEventKeyOrientation value:[self deviceOrientation]];
+    [self updateDictionary:attributes withKey:MGLEventKeyWifi value:@([[MGLReachability reachabilityForLocalWiFi] isReachableViaWiFi])];
+
     return [self eventForAttributes:attributes attributeDictionary:attributeDictionary];
 }
 
 - (MGLMapboxEventAttributes *)mapClickEventWithAttributes:(MGLMapboxEventAttributes *)attributeDictionary {
     MGLMutableMapboxEventAttributes *attributes = [self interactionEvent];
-    attributes[MGLEventKeyEvent] = MGLEventTypeMapTap;
+    [self updateDictionary:attributes withKey:MGLEventKeyEvent value:MGLEventTypeMapTap];
     return [self eventForAttributes:attributes attributeDictionary:attributeDictionary];
 }
 
 - (MGLMapboxEventAttributes *)mapDragEndEventWithAttributes:(MGLMapboxEventAttributes *)attributeDictionary {
     MGLMutableMapboxEventAttributes *attributes = [self interactionEvent];
-    attributes[MGLEventKeyEvent] = MGLEventTypeMapDragEnd;
+    [self updateDictionary:attributes withKey:MGLEventKeyEvent value:MGLEventTypeMapDragEnd];
+
     return [self eventForAttributes:attributes attributeDictionary:attributeDictionary];
 }
 
 - (MGLMutableMapboxEventAttributes *)interactionEvent {
-    MGLMutableMapboxEventAttributes *attributes = [@{MGLEventKeyCreated: [self.rfc3339DateFormatter stringFromDate:[NSDate date]],
-                                                     MGLEventKeyOrientation: [self deviceOrientation],
-                                                     MGLEventKeyWifi: @([[MGLReachability reachabilityForLocalWiFi] isReachableViaWiFi])} mutableCopy];
-    [self addBatteryStateToAttributes:attributes];
```

Comment:
It looks like battery state is no longer added to the attributes and `addBatteryStateToAttributes:` hasn’t been updated to use `updateDictionary:withKey:value:`.